ffs/scheduler/scheduler.go

```diff
@@ -374,15 +395,39 @@ forLoop:
 }
 
 func (s *Scheduler) executeQueuedJob(j ffs.Job) {
+	cancelChan := make(chan struct{})
+	// Create chan to allow Job cancellation.
+	s.cancelLock.Lock()
+	s.cancelChans[j.ID] = cancelChan
+	s.cancelLock.Unlock()
+	defer func() {
+		s.cancelLock.Lock()
+		delete(s.cancelChans, j.ID)
+		s.cancelLock.Unlock()
+	}()
+
+	// Get
 	a, err := s.as.Get(j.ID)
 	if err != nil {
 		log.Errorf("getting push config action data from store: %s", err)
 		return
 	}
 
-	ctx := context.WithValue(s.ctx, ffs.CtxKeyJid, j.ID)
+	ctx, cancel := context.WithCancel(context.WithValue(context.Background(), ffs.CtxKeyJid, j.ID))
+	defer cancel()
+	go func() {
+		// If the user called Cancel to cancel Job execution,
+		// we cancel the context to finish.
+		<-cancelChan
+		cancel()
+	}()
```

Comment:
Someone canceled the Job?. Then cancel its context of execution, so it will stop.